### PR TITLE
Multiple accounts

### PR DESCRIPTION
**Pull request by ErlendHaa**

### CODE REVIEW

File: api/query.go [hunk 1]
```diff
@@ -12,7 +12,8 @@ import (
 )
 
 type BaseRequest struct {
-	// The blob path to a vds in form: container/subpath
+	// The blob url to a vds in form
+	// https://<account>.blob.core.windows.net/<container>/<blob>",
```
Comment (achaikou) [inline api/query.go:16]: ouch. This is how it renders
`https://.blob.core.windows.net//",`

File: cmd/query/main.go [hunk 2]
```diff
@@ -31,7 +32,8 @@ func parseopts() opts {
 		&opts.storageURL,
 		"storage-url",
 		0,
-		"Storage URL, e.g. https://<account>.blob.core.windows.net",
+		"Comma-seperated list of storage accounts that should be accepted by the API. " +
+		"E.g. https://<account>.blob.core.windows.net,...",
```
[the remark this answers is not in the document]
Comment (ErlendHaa) [reply cmd/query/main.go:36]: I've updated the example. Then I added space-stripping the parsing and a guard against empty strings. I would prefer that the server fail to start if the input format is wonky! Good catch

File: internal/vds/vds.go [hunk 3]
```diff
@@ -99,6 +99,33 @@ type Connection struct {
 	Credential string
 }
 
+/*
+ * Sanitize path and make the input path into a *url.URL type
+ *
+ * OpenVDS segfault if the blobpath ends on a trailing slash. We don't want to
+ * propogate that behaviour to the user in any way, so we need to explictly
```
[the remark this answers is not in the document]
Comment (ErlendHaa) [reply internal/vds/vds.go:106]: Unless minor typos like this adds ambiguity to the message, I don't see it as a big deal for these internal dev comments.

File: internal/vds/vds.go [hunk 4]
```diff
@@ -99,6 +99,33 @@ type Connection struct {
...
+ */
+func makeUrl(path string) (*url.URL, error) {
+	if strings.HasSuffix(path, "/") {
+		path = path[:len(path) - 1]
+	}
+
+	return url.Parse(path)
+}
+
+func isWhiteListed(whitelist []*url.URL, requested *url.URL) error {
```
[the remark this answers is not in the document]
Comment (ErlendHaa) [reply internal/vds/vds.go:118]: jeez

File: api/query.go (hunk 1 above)
Comment (ErlendHaa) [reply api/query.go:16]: Good catch!

File: cmd/query/main.go [hunk 5]
```diff
@@ -12,6 +13,7 @@ import (
 
 	_ "github.com/equinor/vds-slice/docs"
 	"github.com/equinor/vds-slice/api"
+	"github.com/equinor/vds-slice/internal/vds"
```
[the remark this answers is not in the document]
Comment (ErlendHaa) [reply cmd/query/main.go:16]: Yes and no! The connection stuff is VDS specific. It's vds-connection. Sanitisation is due to vds being rude with it's segfaults etc...

**Review by achaikou — APPROVED** [review]

**Pull request merged**